core/src/main/cpp/gdn/gast_node.h

```diff
@@ -81,85 +74,55 @@ GODOT_CLASS(GastNode, StaticBody)
         return collidable;
     }
 
-    inline void set_curved(bool curved) {
-        if (this->curved == curved) {
-            return;
-        }
-        this->curved = curved;
-        update_mesh_dimensions_and_collision_shape();
-    }
-
-    inline bool is_curved() {
-        return this->curved;
-    }
-
-    inline void set_projection_mesh_type(int projection_mesh_type) {
-        set_projection_mesh_type(static_cast<ProjectionMeshType>(projection_mesh_type));
+    inline void set_projection_mesh(int projection_mesh_type) {
+        set_projection_mesh(static_cast<ProjectionMesh::ProjectionMeshType>(projection_mesh_type));
     }
 
-    inline void set_projection_mesh_type(ProjectionMeshType projection_mesh_type) {
-        if (this->projection_mesh_type == projection_mesh_type) {
+    inline void set_projection_mesh(ProjectionMesh::ProjectionMeshType projection_mesh_type) {
+        if (projection_mesh_type == projection_mesh->get_projection_mesh_type()) {
             return;
         }
-        this->projection_mesh_type = projection_mesh_type;
-        update_mesh_dimensions_and_collision_shape();
-    }
-
-    inline ProjectionMeshType get_projection_mesh_type() {
-        return this->projection_mesh_type;
-    }
 
-    inline void set_gaze_tracking(bool gaze_tracking) {
-        if (this->gaze_tracking == gaze_tracking) {
-            return;
+        projection_mesh->reset_external_texture();
+
+        if (projection_mesh_type == ProjectionMesh::ProjectionMeshType::RECTANGULAR) {
+            projection_mesh =
+                    projection_mesh_pool.get_or_create_projection_mesh<RectangularProjectionMesh>();
+        } else if (projection_mesh_type == ProjectionMesh::ProjectionMeshType::EQUIRECTANGULAR) {
+            projection_mesh =
+                    projection_mesh_pool.get_or_create_projection_mesh<EquirectangularProjectionMesh>();
+        } else {
+            ALOGE("Projection mesh type %d unimplemented, falling back to RECTANGULAR.",
+                    projection_mesh_type);
+            projection_mesh =
+                    projection_mesh_pool.get_or_create_projection_mesh<RectangularProjectionMesh>();
```

Comment:
Seems like a `switch` would look better here, and you can also combine the `default` case with the `RECTANGULAR` case.
The `default` case would come first so you can still print the error message, then fall-through to the `RECTANGULAR` case.

Reply:
Done.